src/python/pants/backend/docker/target_types.py

```diff
@@ -46,6 +47,24 @@ class DockerBuildArgsField(StringSequenceField):
     )
 
 
+class DockerImageContextRootField(StringField):
+    alias = "context_root"
+    help = (
+        "Specify which directory to use as the Docker build context root, relative to the build "
+        "root by default. With a `./` prefix it is relative to the directory of the BUILD file. "
+        "This affects the file paths to use for the COPY and ADD instructions.\n\nThe "
+        "default build root is specified in the `[docker].default_context_root` configuration "
+        "option."
```

Comment:
How about this?

> Specify which directory to use as the Docker build context root. This affects the file paths to use for the COPY and ADD instructions. For example, whether `COPY f.txt` should look for the file `<build root>/f.txt` vs `<build root/path_to_build_file/f.txt`.
>
> Specify a path like `src/docker`, or use `.` to specify the build root. You can also prefix with `./` to make the value relative to the directory of the BUILD file, such as `./assets`.
>
> If not specified, will default to `docker].default_context_root`.

--

> relative to the build root by default.

That's a lie, I think. The default is whatever the option is.

Reply:
>> relative to the build root by default.

> That's a lie, I think. The default is whatever the option is.

That sentence may need to be rephrased, then. As the option does not apply when you provide a value for the field. What I was referring to is that when you use `"src/project/a"` for the field value, that path is relative to the build root, where as `"./a/subdir"` is relative to the path of the BUILD file.

Reply:
Ah, got it. That makes sense upon rereading.

I think I was confused because of mixing distinct things into the same sentences. In contrast, my above recommendation splits up the content like this:

* P1: what the heck is this option / how is it used?
* P2: what values can I put?
* P3: what happens if I leave it off?

Reply:
OK, rephrased along the lines of your suggestion @Eric-Arellano, with some tweaks of my own. wdyt?